source/jobs/JobsFeature.cpp

```diff
@@ -33,6 +33,8 @@ using namespace Aws::Iot::DeviceClient::Util;
 using namespace Aws::Iot::DeviceClient::Jobs;
 using namespace Aws::Iotjobs;
 
+const std::string JobsFeature::DEFAULT_JOBS_HANDLER_DIR = "~/.aws-iot-device-client/jobs/";
```

Comment:
Assigned the string value in the header file?

Reply:
This was the comment Marco left about it

```
This should be static const std::string instead of const std::string. The difference is that the former creates exactly 1 instance of the variable no matter how many instances of the class are created and the latter creates an instance of the variable for every instance of the class.

Since std::string is a non integral type you will have to split the declaration and definition.

In .h file add static const std::string DEFAULT_JOBS_HANDLER_DIR;
In .cpp file add const std::string JobsFeature::DEFAULT_JOBS_HANDLER_DIR = "...";
```

Reply:
You can only initialize a static const value in the header for integer types, not other types. The second best choice here is to use `static constexpr char[]` or (if c++17) use `static const std::string_view`.